### DISCUSSION TITLE
Artisan / Artisan.plus communication

### DISCUSSION THREAD

**Discussion by ChertFlint** [artisan-roaster-scope/artisan#1751] [Q&A]
Hi,

Thanks to the developers of artisan .

I use artisan with linux mint, a ubuntu iteration. Earlier this year, I noticed that my inventory of beans in artisan.plus is not reliably tracked such that I have had depleted bean stocks but still inventory listed in artisan.plus.  Eventually I corrected the stocks when I ran ouf of green coffee, then updated when I had purchased more.  I started to pay more attention and I always see a blue plus to the upper left when roasting.  Often after 'drop' I see that my roast is logged and auto-saved when I press off.  But some roasts are not logged to artisan.plus or not saved.

This week I used the scheduler function and had a similar problem.  After the first roast, was done, the list decreased by one and the roast moved to 'completed' but then two subsequent roasts did not.  The blue cross was blue.  And the bean mass was not subtracted from Artisan.plus.  Then 3 of the scheduled roasts were subtracted from the list as I moved along my roast plan .    At the end of my 6 roast sessions, I went back and opened the files and saved which then caused them to show up as roasts in Artisan.plus. and to move to completed.

  I assume that  the internet connectedness is not apparent when a trigger save and  it does not work.   But I don'thave any evidence that the internet connection was dropped

This is more of an error report than asking to help resolve the issue, because as a home roaster trackin inventory is a plus but is note required, but if you know a solution to the problem, I am interested.

Thanks

**Comment by MAKOMO** [comment]
Dear Chert,

thanks for your report and your kind words! This one is very interesting to us!

> Earlier this year, I noticed that my inventory of beans in artisan.plus is not reliably tracked such that I have had depleted bean stocks but still inventory listed in artisan.plus.

This does not sound too good. Using an inventory management system only to find out that it does not reliable track your inventory is a downer.

> Often after 'drop' I see that my roast is logged and auto-saved when I press off. But some roasts are not logged to artisan.plus or not saved.

Artisan sends roasts to the artisan.plus platform on DROP. It packs together the metadata of the just completed roast and copies it to the outbox, a persisted queue running in parallel which tries (and retries) to send it to the server. The metadata is also added to the outbox queue the moment you save the profile to an .alog file if it did not succeed before. In case the data got successfully transferred to the server, it gets synchronized with the server on succeeding loads and saves while Artisan is connected to the platform.

> This week I used the scheduler function and had a similar problem. After the first roast, was done, the list decreased by one and the roast moved to 'completed' but then two subsequent roasts did not. The blue cross was blue. And the bean mass was not subtracted from Artisan.plus. Then 3 of the scheduled roasts were subtracted from the list as I moved along my roast plan . At the end of my 6 roast sessions, I went back and opened the files and saved which then caused them to show up as roasts in Artisan.plus. and to move to completed.

Hm. Still doing some forensic work with the data we have, but according to our initial inspection of the server logs and your data you did 7 roasts  in that session (you also scheduled 7 roasts on that day!). The first 2 kind of "failed", the 3rd was correctly registered, the 4th failed again, an the remaining were again registered correctly. The 3 failed roasts you established later on the server by loading them and pressing the red plus icon as you report.

> I assume that the internet connectedness is not apparent when a trigger save and it does not work. But I don'thave any evidence that the internet connection was dropped

The cause of this desaster does not seem to be a problem with the connectivity. It seems also to be independent of your use of the scheduler or not.

Looking at your roasts I see that exactly the ones that you uploaded after the session as they were missing do not have a DROP event marked. Additionally, those roasts do not carry a batch number (only the prefix). As the DROP event did not trigger on those roasts were regarded by Artisan as incomplete and the batch number was not assigned (note that this is independent of the artisan.plus platform).

There is even a mechanism that automatically set DROP to the last moment in the recording if not yet set, but a CHARGE event was set, the roast time since CHARGE is exceeding 7min and autoDROP is active or the DROP button is hidden. Thus even if you forgot to press DROP and directly pressed OFF at the end of your roast, the batch number should have been assigned and the roasts meta information sent to artisan.plus.

Questions:

1) do you have autoDROP ticked under menu Config >> Events, 1st tab? 
2) is your DROP button visible during recording?
3) Would it be possible to share one of those "failed" roasts with us for further investigations?

Note that you can set the DROP manual after the recording via a right-mouse click to the BT curve at the correct position.

THANKS FOR ALL!
Marko

**Comment by ChertFlint** [comment]
Thanks for looking into this Marko.
TO answer your questions.
I cannot attach .alog files here it seems, but I would be happy to share them if it would help. 

 I did not have autoDROP checked so I made that change just now   I do keep my DROP button visible.  I always try to end a roast with DROP, await file save and then OFF and then ON and START for next roast.

I will carry on and see how my schedule and inventory updates go from here.

Oh, my DROP has a serial command to turn my DC fan off.  In case that could be a wrinkle in the system.

Thanks again

Flint

**Reply by MAKOMO** [reply]
Chert,

only completed roasts are regarded worse to be uploaded to plus and roasts not uploaded to plus cannot reduce your stock. A completed roast needs a DROP event. This is to avoid all kind of recordings not being regular roasts to be automatically uploaded to plus. In case you have  autoDROP disabled (default ON) it is your responsibility to ensure that the DROP event is set on regular roasts. That seems to have failed in those 3 cases.

> I will carry on and see how my schedule and inventory updates go from here.

Just check if all roasts got uploaded per session to plus and that bean/blend/store, batch size and yield are correctly set correctly. Note that the scheduler is layered on top of the regular synchronization protocol and thus is not relevant here.

> Oh, my DROP has a serial command to turn my DC fan off. In case that could be a wrinkle in the system.

That serial command for your DC fan does not matter here as this is processed in parallel.

To upload roast profiles here to GitHub you need to rename the files from .alog to end in .txt. Then you can drag-and-drop them here on a comment field for sharing.

THANKS!

**Comment by ChertFlint** [comment]
[artisan-settings.txt](https://github.com/user-attachments/files/17937387/artisan-settings.txt)
[OOE2613 Burundi Burundi Burundi Rubirizi Washe 24-11-21 212 ~37.1 197.3 199.4..txt](https://github.com/user-attachments/files/17937388/OOE2613.Burundi.Burundi.Burundi.Rubirizi.Washe.24-11-21.212.37.1.197.3.199.4.txt)
[OOE2612 Hacea B Mexico Chiapas Finca Nuevo Méx 24-11-21 118 ~22.9 197.6 215.8..txt](https://github.com/user-attachments/files/17937389/OOE2612.Hacea.B.Mexico.Chiapas.Finca.Nuevo.Mex.24-11-21.118.22.9.197.6.215.8.txt)
[OOE2611 Hacea B Mexico Chiapas Finca Nuevo Méx 24-11-21 134 ~24.2 197.1 217.3.txt](https://github.com/user-attachments/files/17937390/OOE2611.Hacea.B.Mexico.Chiapas.Finca.Nuevo.Mex.24-11-21.134.24.2.197.1.217.3.txt)
[OOE0 Guatemala Onyx ALMA blend program Guatemala Onyx ALMA blend prog 24-11-21 0 ~0.0 196.1 0..txt](https://github.com/user-attachments/files/17937391/OOE0.Guatemala.Onyx.ALMA.blend.program.Guatemala.Onyx.ALMA.blend.prog.24-11-21.0.0.0.196.1.0.txt)
[OOE2610 Guatemala Onyx ALMA blend program Guatemala Onyx ALMA blend prog 24-11-21 134 ~23.2 197.2 210.7..txt](https://github.com/user-attachments/files/17937392/OOE2610.Guatemala.Onyx.ALMA.blend.program.Guatemala.Onyx.ALMA.blend.prog.24-11-21.134.23.2.197.2.210.7.txt)
[OOE0 Guatemala Onyx ALMA blend program Guatemala Onyx ALMA blend prog 24-11-21 0 ~0.0 197.8 0..txt](https://github.com/user-attachments/files/17937393/OOE0.Guatemala.Onyx.ALMA.blend.program.Guatemala.Onyx.ALMA.blend.prog.24-11-21.0.0.0.197.8.0.txt)
[OOE0 Brazil 2024, Daterra Masterpiece Borem Brazil Daterra Masterpiece Bor 24-11-21 0 ~0.0 201.9 0..txt](https://github.com/user-attachments/files/17937394/OOE0.Brazil.2024.Daterra.Masterpiece.Borem.Brazil.Daterra.Masterpiece.Bor.24-11-21.0.0.0.201.9.0.txt)


Attached are the files from that roast day along with the .aset

- lots to learn, lots to learn in github
Flint

**Reply by MAKOMO** [reply]
Dear Chert, thanks for attaching those files. They look all good and confirm my guess that those roasts that did not upload automatically lack the DROP event and also did not progress the batch counter. Adding those DROP events manually after the roast via a right click on the position @BT is simple as I said before. Also the batch counter can be added or edited after the roast by right click on the "Batch" label in the RoastProperties.

You are still using v3.0.2. You could upgrade to the just released v3.1. For the next release I will slightly improve the signaling of the upload state based on this discussion.

I quickly looked at your settings too and they look ok to me. Your RoR could be smoother by 

- increasing the TC filter from 10% to 70% (under Devices, TC4). Note that this filter is more effective with minimal delay than anything Artisan can do as it runs on a way faster loop.
- Smooth Curves: 1
- RoR filter: Delta Span to 10 or 15s, Smoothing to 3s (or 0s; off)
- untick Polyfit (that algorithm seems never to outperform the standard algo.)

If you decide to turn autoDrop on, you might want to use the "Sensitive" variant instead of the "Standard" setting as your signal at DROP is not very significant. Thus the "Sensitive" mode is more prone to false positives. Take an ey on this and report!
